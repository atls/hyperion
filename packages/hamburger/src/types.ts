export interface LineProps {
  active: boolean
  color?: string
}

export interface BurgerProps {
  color?: string
}
