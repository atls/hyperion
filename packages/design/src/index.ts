export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
