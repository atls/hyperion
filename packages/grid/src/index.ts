export * from './Column'
export * from './Grid'
