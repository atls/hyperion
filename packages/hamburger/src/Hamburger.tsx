import React                      from 'react'
import { ifProp, switchProp }     from 'styled-tools'

import styled                     from '@emotion/styled'

import { BurgerProps, LineProps } from './types'

const Container = styled.div(() => ({
  cursor: 'pointer',
  top: 0,
  zIndex: 15,
  padding: '28px',
  boxSizing: 'border-box',
  width: '56px',
  height: '56px',
  '&:hover': {
    opacity: 0.6,
  },
}))

const Burger = styled.div<BurgerProps>(
  () => ({
    width: '1.625em',
    height: '1.625em',
    margin: '-0.8125em 0 0 -0.8125em',
    padding: 0,
    top: '50%',
    left: '50%',
    transition:
      'transform 1s cubic-bezier(0.23, 1, 0.32, 1), color 1s cubic-bezier(0.23, 1, 0.32, 1)',
    transform: 'translateZ(0)',
    boxSizing: 'border-box',
  }),
  switchProp('color', () => ({
    alto: {
      color: 'rgb(216, 216, 216)',
    },
    white: {
      color: '#ffffff',
    },
  }))
)

const Line = styled.div<LineProps>(
  () => ({
    width: '100%',
    height: '2px',
    top: '50%',
    position: 'absolute',
    marginTop: '-0.75px',
    transform: 'translateY(-3.75px) translateZ(0)',
    transition:
      'transform 1s cubic-bezier(0.23, 1, 0.32, 1), background-color 1s cubic-bezier(0.23, 1, 0.32, 1)',
    '&:last-child': {
      transform: 'translateY(3.75px) translateZ(0)',
    },
  }),
  ifProp(
    { active: true },
    {
      transform: 'rotate(45deg) translateZ(0)',
      '&:last-child': {
        transform: 'rotate(-45deg) translateZ(0)',
      },
    }
  ),
  switchProp('color', () => ({
    alto: {
      backgroundColor: 'rgb(216, 216, 216)',
    },
    white: {
      backgroundColor: '#ffffff',
    },
  }))
)

export const Hamburger = ({
  color = 'alto',
  active = false,
  onOpen = () => {},
  onClose = () => {},
}) => (
  <Container onClick={active ? onClose : onOpen}>
    <Burger color={color}>
      <Line color={color} active={active} />
      <Line color={color} active={active} />
    </Burger>
  </Container>
)
