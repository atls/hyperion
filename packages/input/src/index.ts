export * from './Textarea'
export * from './Input'
export * from './Checkbox'
