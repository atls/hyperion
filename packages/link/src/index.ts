export * from './Link'
export * from './NextLink'
export * from './RouteLink'
