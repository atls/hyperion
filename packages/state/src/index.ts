export * from './withState'
export * from './useActive'
export * from './useHover'
export * from './useFocus'
