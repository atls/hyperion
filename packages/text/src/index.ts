export * from './Text'
export * from './Space'
export * from './SearchResult'
