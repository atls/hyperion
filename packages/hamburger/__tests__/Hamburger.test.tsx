import React         from 'react'
import renderer      from 'react-test-renderer'

import { Hamburger } from '../src/index'

it('Hamburger should render active&inactive states correctly', () => {
  const treeInactive = renderer
    .create(<Hamburger color='alto' active={false} onClose={null} onOpen={null} />)
    .toJSON()

  const treeActive = renderer
    .create(<Hamburger color='white' active onClose={null} onOpen={null} />)
    .toJSON()

  expect(treeInactive).toMatchSnapshot()
  expect(treeActive).toMatchSnapshot()
})
