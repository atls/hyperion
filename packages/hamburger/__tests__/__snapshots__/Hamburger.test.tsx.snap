// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger should render active&inactive states correctly 1`] = `
.emotion-3 {
  cursor: pointer;
  top: 0;
  z-index: 15;
  padding: 28px;
  box-sizing: border-box;
  width: 56px;
  height: 56px;
}

.emotion-3:hover {
  opacity: 0.6;
}

.emotion-2 {
  width: 1.625em;
  height: 1.625em;
  margin: -0.8125em 0 0 -0.8125em;
  padding: 0;
  top: 50%;
  left: 50%;
  -webkit-transition: -webkit-transform 1s cubic-bezier(0.23, 1, 0.32, 1),color 1s cubic-bezier(0.23, 1, 0.32, 1);
  transition: transform 1s cubic-bezier(0.23, 1, 0.32, 1),color 1s cubic-bezier(0.23, 1, 0.32, 1);
  -webkit-transform: translateZ(0);
  -moz-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  box-sizing: border-box;
  color: rgb(216, 216, 216);
}

.emotion-0 {
  width: 100%;
  height: 2px;
  top: 50%;
  position: absolute;
  margin-top: -0.75px;
  -webkit-transform: translateY(-3.75px) translateZ(0);
  -moz-transform: translateY(-3.75px) translateZ(0);
  -ms-transform: translateY(-3.75px) translateZ(0);
  transform: translateY(-3.75px) translateZ(0);
  -webkit-transition: -webkit-transform 1s cubic-bezier(0.23, 1, 0.32, 1),background-color 1s cubic-bezier(0.23, 1, 0.32, 1);
  transition: transform 1s cubic-bezier(0.23, 1, 0.32, 1),background-color 1s cubic-bezier(0.23, 1, 0.32, 1);
  background-color: rgb(216, 216, 216);
}

.emotion-0:last-child {
  -webkit-transform: translateY(3.75px) translateZ(0);
  -moz-transform: translateY(3.75px) translateZ(0);
  -ms-transform: translateY(3.75px) translateZ(0);
  transform: translateY(3.75px) translateZ(0);
}

<div
  className="emotion-3"
  onClick={null}
>
  <div
    className="emotion-2"
    color="alto"
  >
    <div
      className="emotion-0"
      color="alto"
    />
    <div
      className="emotion-0"
      color="alto"
    />
  </div>
</div>
`;

exports[`Hamburger should render active&inactive states correctly 2`] = `
.emotion-3 {
  cursor: pointer;
  top: 0;
  z-index: 15;
  padding: 28px;
  box-sizing: border-box;
  width: 56px;
  height: 56px;
}

.emotion-3:hover {
  opacity: 0.6;
}

.emotion-2 {
  width: 1.625em;
  height: 1.625em;
  margin: -0.8125em 0 0 -0.8125em;
  padding: 0;
  top: 50%;
  left: 50%;
  -webkit-transition: -webkit-transform 1s cubic-bezier(0.23, 1, 0.32, 1),color 1s cubic-bezier(0.23, 1, 0.32, 1);
  transition: transform 1s cubic-bezier(0.23, 1, 0.32, 1),color 1s cubic-bezier(0.23, 1, 0.32, 1);
  -webkit-transform: translateZ(0);
  -moz-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  box-sizing: border-box;
  color: #ffffff;
}

.emotion-0 {
  width: 100%;
  height: 2px;
  top: 50%;
  position: absolute;
  margin-top: -0.75px;
  -webkit-transform: translateY(-3.75px) translateZ(0);
  -moz-transform: translateY(-3.75px) translateZ(0);
  -ms-transform: translateY(-3.75px) translateZ(0);
  transform: translateY(-3.75px) translateZ(0);
  -webkit-transition: -webkit-transform 1s cubic-bezier(0.23, 1, 0.32, 1),background-color 1s cubic-bezier(0.23, 1, 0.32, 1);
  transition: transform 1s cubic-bezier(0.23, 1, 0.32, 1),background-color 1s cubic-bezier(0.23, 1, 0.32, 1);
  -webkit-transform: rotate(45deg) translateZ(0);
  -moz-transform: rotate(45deg) translateZ(0);
  -ms-transform: rotate(45deg) translateZ(0);
  transform: rotate(45deg) translateZ(0);
  background-color: #ffffff;
}

.emotion-0:last-child {
  -webkit-transform: translateY(3.75px) translateZ(0);
  -moz-transform: translateY(3.75px) translateZ(0);
  -ms-transform: translateY(3.75px) translateZ(0);
  transform: translateY(3.75px) translateZ(0);
}

.emotion-0:last-child {
  -webkit-transform: rotate(-45deg) translateZ(0);
  -moz-transform: rotate(-45deg) translateZ(0);
  -ms-transform: rotate(-45deg) translateZ(0);
  transform: rotate(-45deg) translateZ(0);
}

<div
  className="emotion-3"
  onClick={null}
>
  <div
    className="emotion-2"
    color="white"
  >
    <div
      className="emotion-0"
      color="white"
    />
    <div
      className="emotion-0"
      color="white"
    />
  </div>
</div>
`;
